lib/pbench/test/unit/server/test_datasets_list.py

```diff
@@ -0,0 +1,156 @@
+import datetime
+from http import HTTPStatus
+from typing import List
+
+from freezegun.api import freeze_time
+from pbench.server import PbenchServerConfig
+from pbench.server.database.models.datasets import Dataset
+
+import pytest
+
+import requests
+
+from pbench.server.api.resources import JSON
+
+
+class TestDatasetsList:
+    """
+    Test the `datasets/list` API. We perform a variety of queries using a set
+    of datasets:
+
+        Owner   Access  Date        Name
+        ------- ------- ----------- ---------
+        drb     private 2020-02-15  drb
+        test    private 2002-06-16  test
+        drb     public  2020-02-15  fio_1
+        test    public  2002-06-16  fio_2
```

Comment:
The default `upload` time will be affected when we create a Dataset. The fact that the different freeze-ray scopes have different default just means those datasets weren't created at the same time. (And, yes, I did that deliberately just because it was easy.)

And, yeah, I could vary any of the dates, and considered that, but decided it wasn't useful.